Avoid installing symlink to haddoc when wrapper installation is disabled
diff --git a/utils/haddock/ghc.mk b/utils/haddock/ghc.mk
index cd420e1..f4c9270 100644
--- a/utils/haddock/ghc.mk
+++ b/utils/haddock/ghc.mk
@@ -47,6 +47,8 @@ install_utils/haddock_data:
 	$(foreach i,$(utils/haddock_dist_DATA_FILES), \
 	    $(call make-command,$(INSTALL_DATA) $(INSTALL_OPTS) utils/haddock/$i "$(DESTDIR)$(ghclibdir)/$(dir $i)"))
 
+# no wrapper - no symlink
+ifeq "$(utils/haddock_dist_INSTALL_SHELL_WRAPPER)" "YES"
 ifeq "$(Windows)" "NO"
 install: install_haddock_link
 .PNONY: install_haddock_link
@@ -54,6 +56,7 @@ install_haddock_link:
 	"$(RM)" $(RM_OPTS) "$(DESTDIR)$(bindir)/haddock"
 	$(LN_S) haddock-$(ProjectVersion) "$(DESTDIR)$(bindir)/haddock"
 endif
+endif
 
 BINDIST_EXTRAS += $(addprefix utils/haddock/,$(utils/haddock_dist_DATA_FILES))
 
